OrbitQt/orbitcodeeditor.cpp

```diff
@@ -438,6 +457,47 @@ void OrbitCodeEditor::lineNumberAreaPaintEvent(QPaintEvent* event) {
   }
 }
 
+//-----------------------------------------------------------------------------
+void OrbitCodeEditor::HeatMapAreaPaintEvent(QPaintEvent* event) {
+  QPainter painter(heatMapArea);
+  painter.fillRect(event->rect(), QColor(30, 30, 30));
+
+  //![extraAreaPaintEvent_0]
+
+  //![extraAreaPaintEvent_1]
+  QTextBlock block = firstVisibleBlock();
+  int blockNumber = block.blockNumber();
+  int top = static_cast<int>(
+      blockBoundingGeometry(block).translated(contentOffset()).top());
+  int bottom = top + static_cast<int>(blockBoundingRect(block).height());
+  //![extraAreaPaintEvent_1]
+
+  //![extraAreaPaintEvent_2]
+  while (block.isValid() && top <= event->rect().bottom()) {
+    if (block.isVisible() && bottom >= event->rect().top()) {
+      // % of hits in this function from 0.0 to 1.0
+      const double hit_ratio = line_to_hit_ratio_(blockNumber);
+      // Create a log-scale, values smaller or equal to 1% gets mapped to a
+      // value greater then 0. However, the value 1.0 should still be mapped to
+      // 1.0, thus log(101) is used instead of log(100).
+      double log_hits =
+          std::log(std::ceil(hit_ratio * 100) + 1) / std::log(101);
+      if (log_hits < 0) {
```

Comment:
Is this necessary? If yes, then maybe you also want to clamp at 1?

Maybe:
// Clamp to [0, 1]
log_hits = std::min(std::max(0.0, log_hits), 1.0);

Reply:
Obsolete with sqrt. Thanks for this.